packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1696,6 +1696,60 @@ abstract class ParentDataWidget<T extends ParentData> extends ProxyWidget {
 /// for that inherited widget using [BuildContext.dependOnInheritedWidgetOfExactType]
 /// and then returns the [ThemeData].
 ///
+/// When using the `of` method, the `context` must be a descendant of the
+/// [InheritedWidget], meaning it must be "bellow" the [InheritedWidget] in the
+/// tree.
+///
+/// {@tool snippet}
+///
+/// DO
```

Comment:
"In this example, the context used is the one from the `Builder`, which is a child of the `FrogColor` widget, so this works:"

Reply:
done